// scw-contracts-v1.0.0/typechain/factories/FallbackManagerErrors__factory.ts
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */

import { Signer, utils, Contract, ContractFactory, Overrides } from "ethers";
import { Provider, TransactionRequest } from "@ethersproject/providers";
import type {
  FallbackManagerErrors,
  FallbackManagerErrorsInterface,
} from "../FallbackManagerErrors";

const _abi = [
  {
    inputs: [],
    name: "HandlerCannotBeZero",
    type: "error",
  },
];

const _bytecode =
  "0x6080604052348015600f57600080fd5b50603f80601d6000396000f3fe6080604052600080fdfea2646970667358221220139406ecc724a0fa03227ec5f33be8e4fa543abf3f1189be2f773356659992d564736f6c63430008110033";

export class FallbackManagerErrors__factory extends ContractFactory {
  constructor(
    ...args: [signer: Signer] | ConstructorParameters<typeof ContractFactory>
  ) {
    if (args.length === 1) {
      super(_abi, _bytecode, args[0]);
    } else {
      super(...args);
    }
  }

  deploy(
    overrides?: Overrides & { from?: string | Promise<string> }
  ): Promise<FallbackManagerErrors> {
    return super.deploy(overrides || {}) as Promise<FallbackManagerErrors>;
  }
  getDeployTransaction(
    overrides?: Overrides & { from?: string | Promise<string> }
  ): TransactionRequest {
    return super.getDeployTransaction(overrides || {});
  }
  attach(address: string): FallbackManagerErrors {
    return super.attach(address) as FallbackManagerErrors;
  }
  connect(signer: Signer): FallbackManagerErrors__factory {
    return super.connect(signer) as FallbackManagerErrors__factory;
  }
  static readonly bytecode = _bytecode;
  static readonly abi = _abi;
  static createInterface(): FallbackManagerErrorsInterface {
    return new utils.Interface(_abi) as FallbackManagerErrorsInterface;
  }
  static connect(
    address: string,
    signerOrProvider: Signer | Provider
  ): FallbackManagerErrors {
    return new Contract(
      address,
      _abi,
      signerOrProvider
    ) as FallbackManagerErrors;
  }
}
