// scw-contracts-v1.0.0/typechain/factories/IERC777TokensRecipient__factory.ts
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */

import { Contract, Signer, utils } from "ethers";
import { Provider } from "@ethersproject/providers";
import type {
  IERC777TokensRecipient,
  IERC777TokensRecipientInterface,
} from "../IERC777TokensRecipient";

const _abi = [
  {
    inputs: [
      {
        internalType: "address",
        name: "operator",
        type: "address",
      },
      {
        internalType: "address",
        name: "from",
        type: "address",
      },
      {
        internalType: "address",
        name: "to",
        type: "address",
      },
      {
        internalType: "uint256",
        name: "amount",
        type: "uint256",
      },
      {
        internalType: "bytes",
        name: "data",
        type: "bytes",
      },
      {
        internalType: "bytes",
        name: "operatorData",
        type: "bytes",
      },
    ],
    name: "tokensReceived",
    outputs: [],
    stateMutability: "nonpayable",
    type: "function",
  },
];

export class IERC777TokensRecipient__factory {
  static readonly abi = _abi;
  static createInterface(): IERC777TokensRecipientInterface {
    return new utils.Interface(_abi) as IERC777TokensRecipientInterface;
  }
  static connect(
    address: string,
    signerOrProvider: Signer | Provider
  ): IERC777TokensRecipient {
    return new Contract(
      address,
      _abi,
      signerOrProvider
    ) as IERC777TokensRecipient;
  }
}
