// scw-contracts-v1.0.0/typechain/factories/ModuleManagerErrors__factory.ts
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */

import { Signer, utils, Contract, ContractFactory, Overrides } from "ethers";
import { Provider, TransactionRequest } from "@ethersproject/providers";
import type {
  ModuleManagerErrors,
  ModuleManagerErrorsInterface,
} from "../ModuleManagerErrors";

const _abi = [
  {
    inputs: [
      {
        internalType: "address",
        name: "module",
        type: "address",
      },
    ],
    name: "ModuleAlreadyEnabled",
    type: "error",
  },
  {
    inputs: [
      {
        internalType: "address",
        name: "expectedModule",
        type: "address",
      },
      {
        internalType: "address",
        name: "returnedModule",
        type: "address",
      },
      {
        internalType: "address",
        name: "prevModule",
        type: "address",
      },
    ],
    name: "ModuleAndPrevModuleMismatch",
    type: "error",
  },
  {
    inputs: [
      {
        internalType: "address",
        name: "module",
        type: "address",
      },
    ],
    name: "ModuleCannotBeZeroOrSentinel",
    type: "error",
  },
  {
    inputs: [
      {
        internalType: "address",
        name: "module",
        type: "address",
      },
    ],
    name: "ModuleNotEnabled",
    type: "error",
  },
  {
    inputs: [],
    name: "ModulesAlreadyInitialized",
    type: "error",
  },
  {
    inputs: [],
    name: "ModulesSetupExecutionFailed",
    type: "error",
  },
];

const _bytecode =
  "0x6080604052348015600f57600080fd5b50603f80601d6000396000f3fe6080604052600080fdfea26469706673582212207759df3750356bf10ae86bcf498bfbf386cb22475d8f944aa12575bfd680e9da64736f6c63430008110033";

export class ModuleManagerErrors__factory extends ContractFactory {
  constructor(
    ...args: [signer: Signer] | ConstructorParameters<typeof ContractFactory>
  ) {
    if (args.length === 1) {
      super(_abi, _bytecode, args[0]);
    } else {
      super(...args);
    }
  }

  deploy(
    overrides?: Overrides & { from?: string | Promise<string> }
  ): Promise<ModuleManagerErrors> {
    return super.deploy(overrides || {}) as Promise<ModuleManagerErrors>;
  }
  getDeployTransaction(
    overrides?: Overrides & { from?: string | Promise<string> }
  ): TransactionRequest {
    return super.getDeployTransaction(overrides || {});
  }
  attach(address: string): ModuleManagerErrors {
    return super.attach(address) as ModuleManagerErrors;
  }
  connect(signer: Signer): ModuleManagerErrors__factory {
    return super.connect(signer) as ModuleManagerErrors__factory;
  }
  static readonly bytecode = _bytecode;
  static readonly abi = _abi;
  static createInterface(): ModuleManagerErrorsInterface {
    return new utils.Interface(_abi) as ModuleManagerErrorsInterface;
  }
  static connect(
    address: string,
    signerOrProvider: Signer | Provider
  ): ModuleManagerErrors {
    return new Contract(address, _abi, signerOrProvider) as ModuleManagerErrors;
  }
}
