// scw-contracts-v1.0.0/typechain/factories/SelfAuthorized__factory.ts
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */

import { Signer, utils, Contract, ContractFactory, Overrides } from "ethers";
import { Provider, TransactionRequest } from "@ethersproject/providers";
import type {
  SelfAuthorized,
  SelfAuthorizedInterface,
} from "../SelfAuthorized";

const _abi = [
  {
    inputs: [
      {
        internalType: "address",
        name: "caller",
        type: "address",
      },
    ],
    name: "CallerIsNotSelf",
    type: "error",
  },
];

const _bytecode =
  "0x6080604052348015600f57600080fd5b50603f80601d6000396000f3fe6080604052600080fdfea26469706673582212201d1cdcd3bbe8d81f2ec1b207096f7e127c03a3929663f44844b877b595f718f864736f6c63430008110033";

export class SelfAuthorized__factory extends ContractFactory {
  constructor(
    ...args: [signer: Signer] | ConstructorParameters<typeof ContractFactory>
  ) {
    if (args.length === 1) {
      super(_abi, _bytecode, args[0]);
    } else {
      super(...args);
    }
  }

  deploy(
    overrides?: Overrides & { from?: string | Promise<string> }
  ): Promise<SelfAuthorized> {
    return super.deploy(overrides || {}) as Promise<SelfAuthorized>;
  }
  getDeployTransaction(
    overrides?: Overrides & { from?: string | Promise<string> }
  ): TransactionRequest {
    return super.getDeployTransaction(overrides || {});
  }
  attach(address: string): SelfAuthorized {
    return super.attach(address) as SelfAuthorized;
  }
  connect(signer: Signer): SelfAuthorized__factory {
    return super.connect(signer) as SelfAuthorized__factory;
  }
  static readonly bytecode = _bytecode;
  static readonly abi = _abi;
  static createInterface(): SelfAuthorizedInterface {
    return new utils.Interface(_abi) as SelfAuthorizedInterface;
  }
  static connect(
    address: string,
    signerOrProvider: Signer | Provider
  ): SelfAuthorized {
    return new Contract(address, _abi, signerOrProvider) as SelfAuthorized;
  }
}
