// scw-contracts-v1.0.0/typechain/factories/SelfAuthorizedErrors__factory.ts
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */

import { Signer, utils, Contract, ContractFactory, Overrides } from "ethers";
import { Provider, TransactionRequest } from "@ethersproject/providers";
import type {
  SelfAuthorizedErrors,
  SelfAuthorizedErrorsInterface,
} from "../SelfAuthorizedErrors";

const _abi = [
  {
    inputs: [
      {
        internalType: "address",
        name: "caller",
        type: "address",
      },
    ],
    name: "CallerIsNotSelf",
    type: "error",
  },
];

const _bytecode =
  "0x6080604052348015600f57600080fd5b50603f80601d6000396000f3fe6080604052600080fdfea2646970667358221220626f8f86b8e151d3db3e47806bfa73746b1ebd50081f60a744085df397a2fe5b64736f6c63430008110033";

export class SelfAuthorizedErrors__factory extends ContractFactory {
  constructor(
    ...args: [signer: Signer] | ConstructorParameters<typeof ContractFactory>
  ) {
    if (args.length === 1) {
      super(_abi, _bytecode, args[0]);
    } else {
      super(...args);
    }
  }

  deploy(
    overrides?: Overrides & { from?: string | Promise<string> }
  ): Promise<SelfAuthorizedErrors> {
    return super.deploy(overrides || {}) as Promise<SelfAuthorizedErrors>;
  }
  getDeployTransaction(
    overrides?: Overrides & { from?: string | Promise<string> }
  ): TransactionRequest {
    return super.getDeployTransaction(overrides || {});
  }
  attach(address: string): SelfAuthorizedErrors {
    return super.attach(address) as SelfAuthorizedErrors;
  }
  connect(signer: Signer): SelfAuthorizedErrors__factory {
    return super.connect(signer) as SelfAuthorizedErrors__factory;
  }
  static readonly bytecode = _bytecode;
  static readonly abi = _abi;
  static createInterface(): SelfAuthorizedErrorsInterface {
    return new utils.Interface(_abi) as SelfAuthorizedErrorsInterface;
  }
  static connect(
    address: string,
    signerOrProvider: Signer | Provider
  ): SelfAuthorizedErrors {
    return new Contract(
      address,
      _abi,
      signerOrProvider
    ) as SelfAuthorizedErrors;
  }
}
