// scw-contracts-v1.0.0/typechain/factories/SmartAccountErrors__factory.ts
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */

import { Signer, utils, Contract, ContractFactory, Overrides } from "ethers";
import { Provider, TransactionRequest } from "@ethersproject/providers";
import type {
  SmartAccountErrors,
  SmartAccountErrorsInterface,
} from "../SmartAccountErrors";

const _abi = [
  {
    inputs: [
      {
        internalType: "address",
        name: "smartAccount",
        type: "address",
      },
    ],
    name: "AlreadyInitialized",
    type: "error",
  },
  {
    inputs: [],
    name: "BaseImplementationCannotBeZero",
    type: "error",
  },
  {
    inputs: [
      {
        internalType: "address",
        name: "caller",
        type: "address",
      },
    ],
    name: "CallerIsNotAnEntryPoint",
    type: "error",
  },
  {
    inputs: [
      {
        internalType: "address",
        name: "caller",
        type: "address",
      },
    ],
    name: "CallerIsNotEntryPointOrOwner",
    type: "error",
  },
  {
    inputs: [
      {
        internalType: "address",
        name: "caller",
        type: "address",
      },
    ],
    name: "CallerIsNotOwner",
    type: "error",
  },
  {
    inputs: [
      {
        internalType: "uint256",
        name: "targetTxGas",
        type: "uint256",
      },
      {
        internalType: "uint256",
        name: "gasPrice",
        type: "uint256",
      },
      {
        internalType: "bool",
        name: "success",
        type: "bool",
      },
    ],
    name: "CanNotEstimateGas",
    type: "error",
  },
  {
    inputs: [],
    name: "DelegateCallsOnly",
    type: "error",
  },
  {
    inputs: [],
    name: "EntryPointCannotBeZero",
    type: "error",
  },
  {
    inputs: [],
    name: "ExecutionFailed",
    type: "error",
  },
  {
    inputs: [
      {
        internalType: "address",
        name: "implementationAddress",
        type: "address",
      },
    ],
    name: "InvalidImplementation",
    type: "error",
  },
  {
    inputs: [
      {
        internalType: "address",
        name: "restoredSigner",
        type: "address",
      },
      {
        internalType: "address",
        name: "expectedSigner",
        type: "address",
      },
    ],
    name: "InvalidSignature",
    type: "error",
  },
  {
    inputs: [
      {
        internalType: "address",
        name: "caller",
        type: "address",
      },
    ],
    name: "MixedAuthFail",
    type: "error",
  },
  {
    inputs: [
      {
        internalType: "uint256",
        name: "gasLeft",
        type: "uint256",
      },
      {
        internalType: "uint256",
        name: "gasRequired",
        type: "uint256",
      },
    ],
    name: "NotEnoughGasLeft",
    type: "error",
  },
  {
    inputs: [],
    name: "OwnerCanNotBeSelf",
    type: "error",
  },
  {
    inputs: [],
    name: "OwnerCannotBeZero",
    type: "error",
  },
  {
    inputs: [],
    name: "OwnerProvidedIsSame",
    type: "error",
  },
  {
    inputs: [],
    name: "TokenGasPriceFactorCanNotBeZero",
    type: "error",
  },
  {
    inputs: [
      {
        internalType: "address",
        name: "token",
        type: "address",
      },
      {
        internalType: "address",
        name: "dest",
        type: "address",
      },
      {
        internalType: "uint256",
        name: "amount",
        type: "uint256",
      },
    ],
    name: "TokenTransferFailed",
    type: "error",
  },
  {
    inputs: [],
    name: "TransferToZeroAddressAttempt",
    type: "error",
  },
  {
    inputs: [
      {
        internalType: "uint256",
        name: "destLength",
        type: "uint256",
      },
      {
        internalType: "uint256",
        name: "valueLength",
        type: "uint256",
      },
      {
        internalType: "uint256",
        name: "funcLength",
        type: "uint256",
      },
    ],
    name: "WrongBatchProvided",
    type: "error",
  },
  {
    inputs: [
      {
        internalType: "bytes",
        name: "contractSignature",
        type: "bytes",
      },
    ],
    name: "WrongContractSignature",
    type: "error",
  },
  {
    inputs: [
      {
        internalType: "uint256",
        name: "uintS",
        type: "uint256",
      },
      {
        internalType: "uint256",
        name: "contractSignatureLength",
        type: "uint256",
      },
      {
        internalType: "uint256",
        name: "signatureLength",
        type: "uint256",
      },
    ],
    name: "WrongContractSignatureFormat",
    type: "error",
  },
];

const _bytecode =
  "0x6080604052348015600f57600080fd5b50603f80601d6000396000f3fe6080604052600080fdfea264697066735822122094600977d167e5f782d8912309a44436e2eb53474fbaebd704b2ff755cb82e4664736f6c63430008110033";

export class SmartAccountErrors__factory extends ContractFactory {
  constructor(
    ...args: [signer: Signer] | ConstructorParameters<typeof ContractFactory>
  ) {
    if (args.length === 1) {
      super(_abi, _bytecode, args[0]);
    } else {
      super(...args);
    }
  }

  deploy(
    overrides?: Overrides & { from?: string | Promise<string> }
  ): Promise<SmartAccountErrors> {
    return super.deploy(overrides || {}) as Promise<SmartAccountErrors>;
  }
  getDeployTransaction(
    overrides?: Overrides & { from?: string | Promise<string> }
  ): TransactionRequest {
    return super.getDeployTransaction(overrides || {});
  }
  attach(address: string): SmartAccountErrors {
    return super.attach(address) as SmartAccountErrors;
  }
  connect(signer: Signer): SmartAccountErrors__factory {
    return super.connect(signer) as SmartAccountErrors__factory;
  }
  static readonly bytecode = _bytecode;
  static readonly abi = _abi;
  static createInterface(): SmartAccountErrorsInterface {
    return new utils.Interface(_abi) as SmartAccountErrorsInterface;
  }
  static connect(
    address: string,
    signerOrProvider: Signer | Provider
  ): SmartAccountErrors {
    return new Contract(address, _abi, signerOrProvider) as SmartAccountErrors;
  }
}
